File: Client/src/app/_models/UserParams.ts
import { User } from "./User";

export class UserParams {
    gender: string;
    minAge: number = 18;
    maxAge: number = 99;
    pageNumber: number = 1;
    pageSize: number = 5;

    constructor(user: User) {
        this.gender = user.gender === 'male' ? 'female' : 'male';
    }
}

File: Dating-WebAPI/Helpers/UserParams.cs
namespace Dating_WebAPI.Helpers;
public class UserParams
{
    private const int _maxPageSize = 50;
    public int PageNumber { get; set; } = 1;
    private int _pageSize = 10;
    public int PageSize
    {
        get => _pageSize;
        set => _pageSize = (value > _maxPageSize) ? _maxPageSize : value; 
    }

    public string? CurrentUsername { get; set; }
    public string? Gender { get; set; }
    public int MinAge { get; set; } = 18;
    public int MaxAge { get; set; } = 100;   
    public string OrderBy { get; set; } = "lastActive";
}
